tests/integration/ha_tests/helpers.py

```diff
@@ -185,3 +175,105 @@ async def app_name(ops_test: OpsTest) -> str:
             return app
 
     return None
+
+
+def storage_type(ops_test, app):
+    """Retrieves type of storage associated with an application.
+
+    Note: this function exists as a temporary solution until this issue is resolved:
+    https://github.com/juju/python-libjuju/issues/694
+    """
+    model_name = ops_test.model.info.name
+    proc = subprocess.check_output(f"juju storage --model={model_name}".split())
```

Comment:
there's actually an ops_test function for running juju code that can make this a little cleaner:
```suggestion
    get_storage = await ops_test.juju("storage", f"--model={model_name}")
    result = get_storage[1]
```

You could also add the --format=json flag in there, load it into a python dict and then access what you want programmatically. This suggestion might be overcomplicating it, though.

Reply:
+1 to using `ops_test.juju()`

Reply:
I'll chime in here and argue *in favor* of the current implementation for four reasons:
1. `ops_test.juju` just wraps an async way to call a sub process
2. the output of `--format=json` is way more verbose than the default one, which is very straightforward to parse
3. this is a (temporary?) workaround
4. it's done and it works